documentation/Spring-native-vs-Quarkus.asciidoc

```diff
@@ -0,0 +1,102 @@
+discusses both of these two frameworks and their potential suitability with devonfw.
+
+=== Quarkus
+
+Quarkus is a full-stack, Kubernetes-native Java framework made for JVMs.
+With its container-first-philosophy and its native compilation, Quarkus
+optimizes Java for containers with low memory usage and fast startup
+times.
+
+Quarkus achieves this in the following ways:
+
+* *First Class Support for GraalVM*
+* *Build Time Metadata Processing:* As much processing as possible is
+done at build time, so your application will only contain the classes
+that are actually needed at runtime. This results in less memory usage,
+and also faster startup time, as all metadata processing has already been
+done.
+* *Reduction in Reflection Usage:* Quarkus tries to avoid reflection as much as possible in order to reduce startup time and memory usage.
+* *Native Image Pre Boot:* When running in a native image, Quarkus
+pre-boots as much of the framework as possible during the native image
+build process. This means that the resulting native image has already
+run most of the startup code and serialized the result into the
+executable, resulting in an even faster startup-time.
+
+This gives Quarkus the potential for a great platform for serverless cloud and Kubernetes environments. For more information about Quarkus
+and its support for devonfw please refer to the link:quarkus.asciidoc[Quarkus introduction guide].
+
+=== Spring Native
+====
+[Note]
+Spring Native is still in Beta. The current version of Spring
+Native 0.10.5 is designed to be used with Spring Boot 2.5.6.*
+====
+
+Spring Native provides support for compiling Spring applications to
+native executables using the GraalVM native-image compiler. Using a
+native image provides some key advantages, such as instant startup,
+instant peak performance, and reduced memory consumption. However, there are
+also some drawbacks: Creating a native image is a heavy process that is
+slower than a regular application. A native image also has fewer runtime
+optimizations after its warmup. Furthermore, it is less mature than the
+JVM and comes with some different behaviors.
+
+*Difference in regular JVM vs GraalVM*      
+
+* Static analysis of the application from the main entry point is
+performed at build time.
+* Unused parts are removed at build time.
+* Configuration required for reflection, resources, and dynamic proxies.
+* Classpath is fixed at build time.
+* No class lazy loading: everything shipped in the executables will be
+loaded in memory on startup.
+* Some code will run at build time.
+
+There are https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/Limitations.md[limitations] around some aspects of Java applications that are not fully supported and therefore should be made aware of before using Spring Native.
+
+=== Build time and start time for apps
```

Comment:
Since this is only a benchmark for one specific app and the benchmark for other apps may be quite different, we should say something about the scope of this app

Reply:
@GuentherJulian I don't think that this is necessary, since the ratio of build and start times between apps do not vary that much between Quarkus and Spring Native. This benchmark should only give an example how Spring Native compares to Quarkus Speeds in a general way.